packages/kbn-es-query/src/es_query/es_query_esql.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+// retrieves the index pattern from the aggregate query
+export function getIndexPatternFromESQLQuery(esql?: string): string {
```

Comment:
should we support finding multiple indexes ?

Reply:
For now the endpoint is pretty basic and does not support something like that. I will keep updating this logic as additional features get implemented